components/gitpod-db/src/typeorm/entity/db-code-sync-collection.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Entity, PrimaryColumn } from "typeorm";
+import { TypeORM } from "../typeorm";
+
+@Entity()
+// on DB but not Typeorm: @Index("ind_dbsync", ["_lastModified"])   // DBSync
+export class DBCodeSyncCollection {
+    @PrimaryColumn(TypeORM.UUID_COLUMN_TYPE)
+    userId: string;
+
+    @PrimaryColumn(TypeORM.UUID_COLUMN_TYPE)
+    collection: string;
+}
```

Comment:
I'd recommend adding a deleted column already, to allow for db-sync to delete records.

Reply:
How will you manage size of this table? In other words, how will you identify what to clean up? In any case, it may be useful to include a CreatedTimestamp  on the table to help you track when records came into existence - helps with debugging.

Reply:
> I'd recommend adding a deleted column already, to allow for db-sync to delete records.

:+1: I'll add it, I wasn't sure as I saw other tables don't have it either in `tables.ts`, should all tables that support deleting and are synced need this column?

> it may be useful to include a CreatedTimestamp on the table to help you track when records came into existence

There's already an `_lastModified` column, it's just not declared in the ts file as I don't use it explicitly, should I add it either way?

Reply:
> should all delete operations need to set this value to true, meaning I cannot use .delete() method from typeorm?

Yes, and don't forget to filter `deleted=false` in `select`

> There's already an _lastModified column in the migration file, it's just not declared in the ts file as I don't use it explicitly, should I add it either way?

We need to define `_lastModified` while creating the table. but not necessary to have it in entity 

(This is what I did in db changed PR 
https://github.com/gitpod-io/gitpod/pull/10573/commits/f23c2bb8cddf5384424ebee6d416a7554ee33eed#diff-9ead78f16102163ca590f18a006f2c89ba38fcd9d929362687ee36c6f94acc1f )

Reply:
@mustard-mh so this is a bug in [this line](https://github.com/gitpod-io/gitpod/blob/801de49c1fc521a4c206f41511ddb6cd29c99598/components/gitpod-db/src/typeorm/code-sync-resource-db.ts#L86) added in [this PR](https://github.com/gitpod-io/gitpod/commit/4f1c3922b4ad71b6601382068b8a33d9b7cb2f2b) :smile:

Reply:
> @mustard-mh so this is a bug in [this line](https://github.com/gitpod-io/gitpod/blob/801de49c1fc521a4c206f41511ddb6cd29c99598/components/gitpod-db/src/typeorm/code-sync-resource-db.ts#L86) added in [this PR](https://github.com/gitpod-io/gitpod/commit/4f1c3922b4ad71b6601382068b8a33d9b7cb2f2b) 😄

😇